### PR TITLE
CLN: simplify+annotate _shallow_copy

### PR DESCRIPTION
**Pull request by jbrockmendel**
IntervalIndex is still an outlier, will be handled separately

### CODE REVIEW

File: pandas/core/indexes/range.py [hunk 1]
```diff
@@ -385,13 +387,13 @@ def tolist(self):
...
+    def _shallow_copy(self, values=None, name: Label = no_default):
+        name = self.name if name is no_default else name
+
         if values is None:
-            name = kwargs.get("name", self.name)
             return self._simple_new(self._range, name=name)
         else:
-            kwargs.setdefault("name", self.name)
-            return self._int64index._shallow_copy(values, **kwargs)
+            return Int64Index._simple_new(values, name=name)
```
Comment (simonjayhawkins) [inline pandas/core/indexes/range.py]: why is self._int64index replaced with Int64Index?  does this affect subclassing?

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

a few inconsistencies between the methods. such as `name = self.name if name is no_default else name` vs `name = name if name is not lib.no_default else self.name` and `name: Label = no_default` vs `name: Label = lib.no_default` but not blocker.

**Comment by jbrockmendel** [comment]
> a few inconsistencies between the methods

Yah, that's just a matter of which modules already have `lib` imported.

File: pandas/core/indexes/range.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/range.py]: Using Int64Index is more direct, avoids constructing self._int64index

Shouldn't affect subclasses, which isn't really supported anyway.

**Comment by jreback** [comment]
thanks

**Pull request merged**